src/librustc/ty/layout.rs

```diff
@@ -598,7 +620,10 @@ impl<'a, 'tcx> LayoutCx<'tcx, TyCtxt<'a, 'tcx, 'tcx>> {
                 let mut unit = univariant_uninterned(&[], &ReprOptions::default(),
                   StructKind::AlwaysSized)?;
                 match unit.abi {
-                    Abi::Aggregate { ref mut sized } => *sized = false,
+                    Abi::Aggregate { ref mut sized, ref mut has_vla } => {
+                        *sized = false;
+                        *has_vla = false;
+                    }
```

Comment:
Preexisting, but I’m having a hard time understanding what this code wants to do & why Aggregate is not just constructed directly...